internal/Pattern.h

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include <re2/re2.h>
+#include <memory>
+#include <string>
+
+namespace uap_cpp {
+
+class Match;
+
+/**
+ * Wrapper around a re2 regular expression
+ */
+class Pattern {
+ public:
+  Pattern();
+  Pattern(const std::string&, bool caseSensitive = true);
+
+  void assign(const std::string&, bool caseSensitive = true);
+
+  bool match(const std::string&, Match&) const;
+
+ private:
+  std::unique_ptr<re2::RE2> regex_;
+  int groupCount_;
+};
+
+/**
+ * Intended to be used with thread_local to avoid initialization cost
+ */
+class Match {
+ public:
+  Match();
+
+  int size() const;
+  const std::string& get(int index) const;
+
+ private:
+  friend class Pattern;
+  enum { MAX_MATCHES = 10 };
```

Comment:
Sure. I initially tried to stick to C++03, because I didn't know what kind of backward compatible support the library needed (judging by the `-std=c++0x` in the Makefile).